CommonMark/Parser/InlineMethods.cs

```diff
@@ -953,7 +955,14 @@ static Reference ParseLinkDetails(Subject subj)
 
             // rollback the subject position because didn't match anything.
             subj.Position = endlabel;
-            return null;
+            if (supportPlaceholderBrackets)
+            {
+                return new Reference() { IsPlaceholder = true };
```

Comment:
It seems I should set the `Url` property from `subj.Buffer.Substring(subj.LastPendingInline.StartPosition, subj.Position - subj.LastPendingInline.StartPosition - 1)` at that point. Changed.